google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -442,4 +448,75 @@ public void testFailedBatchWithRetry() {
     }
     verify(batchMock, times(6)).submit();
   }
+
+  @Test
+  public void testBatchRetry() {
+    final BlobId blobId1 = BlobId.of(BUCKET_NAME1, BLOB_NAME1);
+    final BlobId blobId2 = BlobId.of(BUCKET_NAME2, BLOB_NAME2);
+    options =
+        StorageOptions.newBuilder()
+            .setProjectId("projectId")
+            .setClock(TIME_SOURCE)
+            .setServiceRpcFactory(rpcFactoryMock)
+            .setRetrySettings(
+                ServiceOptions.getDefaultRetrySettings().toBuilder().setMaxAttempts(5).build())
+            .build();
+    RpcBatch batchMock = mock(RpcBatch.class);
+    when(storageRpcMock.createBatch()).thenReturn(batchMock);
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) throws Throwable {
+                RpcBatch.Callback<StorageObject> callback = invocation.getArgument(1);
+                callback.onSuccess(blobId1.toPb());
+                return null;
+              }
+            })
+        .when(batchMock)
+        .addGet(
+            Mockito.eq(blobId1.toPb()),
+            Mockito.<RpcBatch.Callback<StorageObject>>any(),
+            Mockito.<StorageRpc.Option, Object>anyMap());
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) throws Throwable {
+                RpcBatch.Callback<StorageObject> callback = invocation.getArgument(1);
+                GOOGLE_JSON_ERROR.setCode(503);
+                GOOGLE_JSON_ERROR.setMessage("message");
+                callback.onFailure(GOOGLE_JSON_ERROR);
+                return null;
```

Comment:
Please use variables for '503' and 'message'.

Reply:
Done.